app/models/katello/product_content.rb

```diff
@@ -0,0 +1,28 @@
+module Katello
+  class ProductContent < Katello::Model
+    belongs_to :product, :class_name => 'Katello::Product', :foreign_key => 'product_id', :inverse_of => :product_contents
+    belongs_to :content, :class_name => 'Katello::Content', :foreign_key => 'content_id', :inverse_of => :product_content
+
+    default_scope { includes(:content) }
+
+    validates :content_id, :presence => true, :uniqueness => { :scope => :product_id }
+
+    validates_with Katello::Validators::ProductContentValidator
+
+    delegate :content_type, to: :content
+
+    scope :displayable, -> {
+      joins(:content).where.not("#{content_table_name}.content_type IN (?)", Katello::Repository.undisplayable_types)
+      .order("LOWER(#{content_table_name}.name) ASC")
+    }
+
+    def self.content_table_name
+      Katello::Content.table_name
+    end
+
+    # used by Katello::Api::V2::RepositorySetsController#index
+    def repositories
```

Comment:
Will look into making this a proper association.

Reply:
So, not an association, but cleaned up the approach of getting the right repos, and no longer caching them on the object.